modules/service/src/test/scala/lucuma/odb/graphql/mutation/updateObservationsTimes.scala

```diff
@@ -0,0 +1,136 @@
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+package mutation
+
+import cats.effect.IO
+import cats.syntax.all.*
+import io.circe.Json
+import io.circe.literal.*
+import lucuma.core.model.Target
+import lucuma.core.model.User
+
+class updateObservationsTimes extends OdbSuite
+                            with UpdateConstraintSetOps {
+
+  val pi: User    = TestUsers.Standard.pi(nextId, nextId)
+  val pi2: User   = TestUsers.Standard.pi(nextId, nextId)
+  val staff: User = TestUsers.Standard.staff(nextId, nextId)
+
+  override lazy val validUsers: List[User] = List(pi, pi2, staff)
+
+  private def oneUpdateTest(
+    user:     User,
+    update:   String,
+    query:    String,
+    expected: Either[String, Json]
+  ): IO[Unit] =
+
+    for
+      pid <- createProgramAs(user)
+      oid <- createObservationAs(user, pid)
+      _   <- updateObservationTimes(user, oid, update, query, expected)
+    yield ()
+
+  private def multiUpdateTest(
+    user:    User,
+    updates: List[(String, String, Either[String, Json])]
+  ): IO[Unit] =
+
+    for
+      pid <- createProgramAs(user)
+      oid <- createObservationAs(user, pid)
+      _   <- updates.traverse_ { case (update, query, expected) =>
+        updateObservationTimes(user, oid, update, query, expected)
+      }
+    yield ()
+
+  test("observation time: set") {
+
+    val update   = """
+      observationTime: "2022-08-30 17:18:00"
+    """
+
+    val query    = "observations { observationTime }"
+
+    val expected = json"""
+      {
+        "updateObservationsTimes": {
+          "observations": [
+            {
+              "observationTime": "2022-08-30 17:18:00"
+            }
+          ]
+        }
+      }
+    """.asRight
+
+    oneUpdateTest(pi, update, query, expected)
+  }
+
+  test("observation time: set ISO-8601") {
+
+    val update = """
+      observationTime: "2022-08-30 17:18:00"
+    """
+
+    val query = "observations { observationTime }"
+
+    val expected = json"""
+      {
+        "updateObservationsTimes": {
+          "observations": [
+            {
+              "observationTime": "2022-08-30 17:18:00"
+            }
+          ]
+        }
+      }
+    """.asRight
+
+    oneUpdateTest(pi, update, query, expected)
+  }
+
+  test("observation time: delete") {
+
+    val update0  = """
+      observationTime: "2022-08-30 17:18:00"
+    """
+
+    val update1 = """
+      observationTime: null
+    """
+
+    val query    = "observations { observationTime }"
+
+    val expected0 = json"""
+      {
+        "updateObservationsTimes": {
+          "observations": [
+            {
+              "observationTime": "2022-08-30 17:18:00"
+            }
+          ]
+        }
+      }
+    """.asRight
+
+    val expected1 = json"""
+      {
+        "updateObservationsTimes": {
+          "observations": [
+            {
+              "observationTime": null
```

Comment:
It occurred to me that setting the observation time to the same value for multiple observations might be something that is never really done, but it does seem handy to be able to wipe them all out in one mutation like this.

Reply:
That's right. it tests the machinery of multiple observations though